etna/models/nn/chronos/base.py

```diff
@@ -0,0 +1,277 @@
+        if max_context_size <= 0:
+            raise ValueError("Dataset doesn't have any context timestamps.")
+
+        if max_context_size < self.context_size:
+            warnings.warn("Actual length of a dataset is less that context size. All history will be used as context.")
+        available_context_size = min(max_context_size, self.context_size)
+
+        target = ts.df.loc[:, pd.IndexSlice[:, "target"]]
+        context = torch.tensor(target.values.T[:, :available_context_size])
+
+        if prediction_interval:
+            quantiles_forecast, target_forecast = self.pipeline.predict_quantiles(
+                context=context,
+                prediction_length=prediction_size,
+                quantile_levels=quantiles,
+                **predict_kwargs,
+            )  # shape [n_segments, prediction_length, n_quantiles], [n_segments, prediction_length]
+        else:
+            quantiles_forecast, target_forecast = self.pipeline.predict_quantiles(
+                context=context,
+                prediction_length=prediction_size,
+                **predict_kwargs,
+            )  # shape [n_segments, prediction_length, n_quantiles], [n_segments, prediction_length]
+
+        end_idx = len(ts.index)
+        future_ts = ts.tsdataset_idx_slice(start_idx=end_idx - prediction_size, end_idx=end_idx)
+
+        if prediction_interval:
+            quantiles_predicts = (
+                quantiles_forecast.numpy().transpose(1, 0, 2).reshape(prediction_size, -1)
+            )  # shape [prediction_length, segments * n_quantiles]
+            quantile_columns = [f"target_{quantile:.4g}" for quantile in quantiles]
+            columns = pd.MultiIndex.from_product([ts.segments, quantile_columns], names=["segment", "feature"])
+            quantiles_df = pd.DataFrame(quantiles_predicts[: len(ts.df)], columns=columns, index=future_ts.df.index)
+
+            future_ts.add_prediction_intervals(prediction_intervals_df=quantiles_df)
+
+        future_ts.df.loc[:, pd.IndexSlice[:, "target"]] = target_forecast.numpy().transpose(1, 0)
+
+        return future_ts
+
+    @staticmethod
+    @abstractmethod
+    def list_models() -> List[str]:
+        """
+        Return a list of available pretrained chronos models.
+
+        Returns
+        -------
+        :
+            List of available pretrained chronos models.
+        """
+        pass
+
+    def save(self, path: Path):
+        """Save the model. This method doesn't save model's weights.
+         During ``load`` weights are loaded from the path where they were saved during ``init``
+
+        Parameters
+        ----------
+        path:
+            Path to save object to.
+        """
+        self._save(path=path, skip_attributes=["pipeline"])
+
+    @classmethod
+    def load(cls, path: Path):
+        """Load the model.
+
+        Parameters
+        ----------
+        path:
+            Path to load object from.
+        """
+        return super().load(path=path)
```

Comment:
Are we sure it loads the model from `cache_dir`?

Reply:
It loads model from `cache_dir` not every time, for example with local path
In other cases I don't really know how to check this. But the logic looks good.

Reply:
I only want to point out that `super().load()` doesn't call a constructor. So, I don't really understand how `self.pipeline` should be recreated within `load` method.